docker-compose.yml

```diff
@@ -0,0 +1,24 @@
+version: '3.4'
+
+services:
+  webui:
+    image: ${DOCKER_REGISTRY-}webui
+    build:
+      context: .
+      dockerfile: src/WebUI/Dockerfile
+    environment:
```

Comment:
The idea is that `appsettings.json` still points to `localhost` so the application can work without Docker support. In reality, when hosting containerised applications in K8s for example, you will still need to amend connection strings via Secrets, Environment Variables and ConfigMaps. It is very common for Docker applications to have overrides.

What do you think?

Reply:
My opinion on config: appsettings.json is a great place to put "safe" values (values that are safe to put in source control) but you need to support secrets provided from elsewhere - usually via Environment variables. Environemnt variables fed to docker could be sourced from Azure Key vault, for example.

My Approach:

**appsettings.json**
All variables that are common everywhere
No secrets
where there are settings that you expect to be fulfilled from elsewhere, I find it useful to document these by putting the value names in but leaving the value empty.

**appsettings.{env}.json** 
Safe settings that are specific to an environment. This file is checked in

**appsettings.local.json**
Secret values (keys, connection strings) for the local user. 
I explicitly add support for this to Program.cs

```
 public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .ConfigureAppConfiguration((ctx, builder) =>
                    builder.AddJsonFile("appsettings.local.json", true, false)
                )
                .UseStartup<Startup>();

```
this file is .gitignored so that your personal secrets don't get shared.
I generally find this approach easier to work with than using local secrets manager

**runtime secrets**
these should not be checked in to any source control and are usually fed to the running application in docker via environment variables.

Reply:
Yes, I agree with this approach.